### PR TITLE
Explain circumstances of applying early

### PR DESCRIPTION
**Pull request by rudi-c**

### CODE REVIEW

File: docs/1-Applying/when_to_apply.md [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+
+> Sidenote: Why do companies hire so early? This is a case of "market unraveling", discussed by economic Alvin Roth in "Who Gets What, and Why". In matching markets, there is an incentive for both job seekers and job providers start "trading" (accept an offer) just a little before the market "opens" (recruiting season) to have the best chance of getting the best deal. Unfortunately, this means that the recruiting season moves earlier and early (until it hits the starts of the school year, where it can't really move earlier than that).
+
+### Fall/Winter internships
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: These headings are great!

I think this heading could be "internships at non traditional times" or something, since it could be overlapping seasons. I'd also like to give a shoutout to the southern hemipshere

> If you're applying for internships outside of the traditional summer timeline (or you live on the side of the world where summer isn't May-August but want to do your summer internship in the northern hemisphere), 4 months before is probably plenty of time. 

I usually forget that seasons don't mean month timeline and it's nice to bring that to people's attention ^___^

File: docs/1-Applying/when_to_apply.md [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
 
 If you're applying for internships in the fall or winter, 4 months before is probably plenty of time. If you're a Waterloo student and have an internship for 4 months then school for 4 months, and then another internship - I'd recommend applying during the internship and not during school. Interviews during midterms are just too much. It's an amazing feeling to already have a job while watching all your peers stress out about job apps ;D
+
+One thing to keep in mind is that companies rarely have fall and winter intern positions posted on their website. You should ask friends for referrals, send out emails, etc.
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: > One thing to keep in mind is that companies rarely have intern positions posted on their website outside of standard intern season. But that doesn't necessarily mean they're not interested! You should ask friends for referrals, send out emails to people that work there, etc.

File: docs/1-Applying/when_to_apply.md [hunk 3]
```diff
@@ -1,9 +1,23 @@
 ## When to apply
 
-If you're applying for summer internship and want to go to a big or well known company, often you'll need to apply September/October-ish the year before. Yes, this is early. Yes, this is too early. Yes, it's ridiculous. Pretty much, companies figure that if they extend you an offer before everyone else, you'll be more likely to work there. But then everywhere has been doing this and then if you wait until January the intern class will be full.
+### Summer internships
 
-However, there are also a lot of companies that will keep hiring in the new year, and some that only start hiring in the new year.
+If you're applying for summer internship you need to look into applying in September/October-ish the year before, which is the start of the recruiting season. Yes, this is early. Yes, this is too early. Yes, it's ridiculous. The more you wait, the more likely slots will get filled up. That's why we recommend applying when recruiting season start. If you wait until January, the intern class could be full.
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: why'd you take out the big or well known company bit? you changed the wording to sound like you always have to apply early, but lots of other companies hire much later, and if your goal is a small company it will be hard to wait on top of offers months before you can even apply to the companies you really want

also fix my typo pls :3 "If you're applying for summer internship*s,* you need to look.."

Also - That's why we recommend applying when recruiting season start**s**

File: docs/1-Applying/when_to_apply.md [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+
+> Sidenote: Why do companies hire so early? This is a case of "market unraveling", discussed by economic Alvin Roth in "Who Gets What, and Why". In matching markets, there is an incentive for both job seekers and job providers start "trading" (accept an offer) just a little before the market "opens" (recruiting season) to have the best chance of getting the best deal. Unfortunately, this means that the recruiting season moves earlier and early (until it hits the starts of the school year, where it can't really move earlier than that).
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: cute blurb ^_^

> earlier and **earlier**

I think it's a little unclear still how earlier means you get the best deal. I put it in plain English in the old way I phrased it, but maybe you can bring a bit more of that here. The idea being that if you give an offer before anyone else and are "good enough", they're more likely to take you

File: docs/1-Applying/when_to_apply.md [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+- *Startups* (check out [AngelList](https://angel.co/)). They could be the opposite. Their business is very uncertain and their hiring capacity, needs, projects, business goals, etc, tend to change randomly and quickly. They might only be able to know for sure if they can afford an intern when it's a month or two before the internship would start. While not all startups hire interns, it's worth reaching out to them. We often hear stories of students getting a job at a cool startup at the last minute, right after a slot opened up because the startup got funding (also at the last minute).
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: Period after the header

File: docs/1-Applying/when_to_apply.md [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
+- *Smaller companies*. There are many respectable companies, tech or non-tech, that only have a few dozen employees and can only hire a few interns. They don't have the resources to go campus recruiting and prefer to avoid competing head-on with big-name companies. You may have to send out a lot of emails but many of them will hire later. A lot of these companies might be local, so it's useful to search for tech companies in particular cities.
+- *Non-tech companies*. Similarly, there are many companies that hire interns for programming positions but don't have the prestige of large tech companies. Think banks for example. They might even have a lot of internships slots, but they won't get filled as quickly. Keep in mind, just because a company isn't famous in tech doesn't mean it doesn't have good learning opportunities!
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: > just because a company isn't famous in tech doesn't mean it doesn't have good learning opportunities!

omg I want to make a whole file just about this lol

File: docs/1-Applying/when_to_apply.md [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
+- *"We don't have a fixed number of slots"*. Some companies will say that, and they are probably telling the truth. However, the thing you have to keep in mind is that no company has an unlimited capacity to hire interns. Even if they had an unlimited monetary budget, they still need to find full-time employees capable and willing to be intern mentors, as well as projects to give interns. For that reason, every additional intern pass a certain number will strain their capacity to provide a good internship experience. I've been to a company that hires doesn't have an intern quota, but they become more and more picky over time. The unlimited aspect is really for "just in case an amazing candidate shows up at the last minute".
```
Comment (evykassirer) [inline docs/1-Applying/when_to_apply.md]: pass -> past

**Review by evykassirer — COMMENTED** [review]
I love the detail! thanks so much for all these changes 💓

File: docs/1-Applying/when_to_apply.md (hunk 3 above)
Comment (rudi-c) [reply docs/1-Applying/when_to_apply.md]: Uh good question. I might have been thinking something like "at least look into applying so you don't miss out or have more time to know what you want and prepare for it" but I'd need to specify that and say that "you don't necessarily need to actually apply that early except for big companies". Or I could just add the big/well-known company clause back in.

Comment (evykassirer) [reply docs/1-Applying/when_to_apply.md]: yeah adding the clause back is my preference

**Comment by rudi-c** [comment]
Changes
- Clarify market unraveling
- Add bit about southern hemisphere
- Small fixes

**Comment by evykassirer** [comment]
changes look good - just two small nit things and we can merge :D

**Comment by rudi-c** [comment]
:+1: done
